app/models/manageiq/providers/google/network_manager/refresh_parser.rb

```diff
@@ -22,13 +27,96 @@ def ems_inv_to_hashes
         get_security_groups
         get_network_ports
         get_floating_ips
+        get_load_balancers
+        get_load_balancer_pools
+        get_load_balancer_listeners
+
         _log.info("#{log_header}...Complete")
 
         @data
       end
 
+      # Parses a port range returned by GCP from a string to a Range. Note that
+      # GCP treats the empty port range "" to mean all ports; hence this method
+      # returns 0..65535 when the input is the empty string.
+      #
+      # @param port_range [String] the port range (e.g. "" or "80-123" or "11")
+      # @return [Range] a range representing the port range
+      def self.parse_port_range(port_range)
+        # Three forms:
+        # "80"
+        # "5000-5010"
+        # "" (all ports)
+        m = /\A(\d+)(?:-(\d+))?\Z/.match(port_range)
+        return 0..65_535 unless m
+
+        start = Integer(m[1])
+        finish = m[2] ? Integer(m[2]) : start
+        start..finish
+      end
+
+      # Parses a VM's self_link attribute to extract the project name, zone, and
+      # instance name. Used when other services refer to a VM by its link.
+      #
+      # @param vm_link [String] the full url to the vm (e.g.
+      #   "https://www.googleapis.com/compute/v1/projects/myproject/zones/us-central1-a/instances/foobar")
+      # @return [Hash{Symbol => String}, nil] a hash containing extracted components
+      #   for `:project`, `:zone`, and `:instance`, or nil if the link did not
+      #   match.
+      def self.parse_vm_link(vm_link)
+        link_regexp = %r{\Ahttps://www\.googleapis\.com/compute/v1/projects/([^/]+)/zones/([^/]+)/instances/([^/]+)\Z}
+        m = link_regexp.match(vm_link)
+        return nil if m.nil?
+
+        {
+          :project  => m[1],
+          :zone     => m[2],
+          :instance => m[3]
+        }
+      end
+
       private
 
+      # Lookup a VM in fog via its link to get the VM id (which is equivalent to
+      # the ems_ref).
+      #
+      # @param link [String] the full url to the vm
+      # @return [String, nil] the vm id, or nil if it could not be found
+      def get_vm_id_from_link(link)
+        parts = self.class.parse_vm_link(link)
+        unless parts
+          _log.warn("Unable to parse vm link: #{link}")
+          return nil
+        end
+
+        # Ensure our connection is using the same project; if it's not we can't
+        # do much
+        return nil unless @connection.project == parts[:project]
```

Comment:
Is this a realistic concern?  Can you have a load balancer with instances in another project?